.github/workflows/post-commit.yml

```diff
@@ -0,0 +1,94 @@
+# A workflow to run against renovate-bot's PRs,
+# such as `make package` after it updates the package.json and package-lock.json files.
+
+# The potentially untrusted code is first run inside a low-privilege Job, and the diff is uploaded as an artifact.
+# Then a higher-privilege Job applies the diff and pushes the changes to the PR.
+# It's important to only run this workflow against PRs from trusted sources, after also reviewing the changes!
+
+# There have been vulnerabilities with using `git apply` https://github.blog/2023-04-25-git-security-vulnerabilities-announced-4/
+# At this point a compromised git binary cannot modify any of this repo's branches, only the PR fork's branch,
+# due to our branch protection rules and CODEOWNERS.
+# It aslso cannot submit a new release or modify exsiting releases due to tag protection rules.
```

Comment:
Is there any reason this needs to run on the `slsa-framework/slsa-verifier` repo at all? could contributors just run it manually on their fork? That would somewhat limit exposure on our side and reduces toil for reviewers (i.e. if the dist check fails we could just tell them "run this" and point to the doc on how to run it in their fork. better yet, the error from that check could tell them).

 AFAICT, this would just checkout the forked PR and push to the appropriate branch on the forked repo so wouldn't need to interact with (and thus need permissions for) the main repo at all.

Also makes it easier for us to audit since we would basically just makes sure there aren't any runs for this action on the main repo

Reply:
This is mainly intended to run on PRs from renovate-bot's fork. We wouldn't be able to invoke their fork's workflows. 

Example:
 - https://github.com/slsa-framework/slsa-verifier/pull/717
 - https://github.com/renovate-bot/slsa-verifier/tree/renovate/npm
 - https://github.com/renovate-bot/slsa-verifier/actions/workflows/pre-submit.actions.yml